### ISSUE TITLE
React.PureComponent and react/prefer-stateless-function

### ISSUE THREAD

**Issue by kassiansun** [yannickcr/eslint-plugin-react#777]
PureComponent will be warned with react/prefer-stateless-function like:
```javascript
class App extends React.PureComponent {
    render() {
        return (<h2>Hello</h2>);
    }
}
```

**Comment by ljharb** [comment]
You should be preferring a stateless function there, since there's no state. "PureComponent" usage doesn't affect this rule by design.

**Comment by kassiansun** [comment]
@ljharb So how to make this function stateless?

**Comment by ljharb** [comment]
```js
function App() {
  return (<h2>Hello</h2>);
}
```

**Comment by kassiansun** [comment]
@ljharb Thanks for your tips! I misunderstood 'pure' and 'stateless' for a long time.
